### PR TITLE
Feature/dd node wait for state

### PR DESCRIPTION
**Pull request by jadunham1**
Adding in more node states for dimension data with fall back to the old method if we can't match
Adding tests for node states
Making it so Node objects wait_for_state looks at state instead of status
Add test for Node object in wait for state

### CODE REVIEW

File: libcloud/compute/types.py [hunk 1]
```diff
@@ -222,25 +222,31 @@ class NodeState(Type):
     Standard states for a node
 
     :cvar RUNNING: Node is running.
+    :cvar STARTING: Node is starting up.
     :cvar REBOOTING: Node is rebooting.
     :cvar TERMINATED: Node is terminated. This node can't be started later on.
+    :cvar STOPPING: Node is currently trying to stop.
```
Comment (Kami) [inline libcloud/compute/types.py:228]: I'm fine with adding new starting and stopping state, but not too sure about reconfiguring.

I know some load balancer as a service offerings expose a similar state, but I'm not aware of too many compute providers. In any case, not a big deal.

**Comment by Kami** [comment]
Thanks.

The changes look good to me, but I will wait for @tonybaloney.

File: libcloud/compute/types.py (hunk 1 above)
Comment (jadunham1) [inline libcloud/compute/types.py:228]: I was up in the air on it too, thought about maybe using PENDING for a similar effect. It won't affect my waiting for state which is what this change is trying to bring. I was just trying to be as specific as possible on node states.
I'll wait for consensus from @tonybaloney

File: libcloud/compute/drivers/dimensiondata.py [hunk 2]
```diff
@@ -42,6 +42,32 @@
...
+        NodeState.STOPPING,
+    ('PENDING_CHANGE', 'true', 'POWER_OFF_SERVER'):
+        NodeState.STOPPING,
+    ('PENDING_CHANGE', 'true', 'REBOOT_SERVER'):
+        NodeState.REBOOTING,
+    ('PENDING_CHANGE', 'true', 'RESET_SERVER'):
+        NodeState.REBOOTING,
+    ('PENDING_CHANGE', 'true', 'RECONFIGURE_SERVER'):
+        NodeState.RECONFIGURING,
+}
```
Comment (tonybaloney) [inline libcloud/compute/drivers/dimensiondata.py:69]: Full state list is "NORMAL",  "PENDING_ADD", "PENDING_CHANGE", "PENDING_DELETE",  "FAILED_ADD", "FAILED_CHANGE", "FAILED_DELETE" and  "REQUIRES_SUPPORT". 
REQUIRES_SUPPORT is my favorite. Can you map those too please

**Comment by tonybaloney** [comment]
Add some extra states to the list (optional). But otherwise @jadunham1 
thumbs up

**Comment by jadunham1** [comment]
As I can't test all of these in what they look like (do they have workflows associated ect?) I'd rather push as is and people can add to define states.  Things like failed adds I'm not sure I can simulate correctly.
We're definitely in a better position to react to new states ect in this model.

**Comment by tonybaloney** [comment]
Agreed. :+1:

**Pull request closed without merging**